### PR TITLE
Add update env variable action

### PR DESCRIPTION
**Pull request by HenriChabrand**
## Description

Adding new action allowing user to easily update project env variables.

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @gleich 

## Screencast

Screenshot 2021-10-31 at 21 51 50
Screenshot 2021-10-31 at 21 52 10

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Comment by gleich** [comment]
Hey @HenriChabrand! Thank you so much for adding this to vercast!! I should be able to review this sometime this week. Today I'm a bit busy with some Halloween stuff 🎃.

**Comment by HenriChabrand** [comment]
Hey @gleich ! Sure thing 👻
I tried to mimic as mush as possible your original code, but let me know if I need to make some adjustments 🙌
Have a good one !

**Comment by pernielsentikaer** [comment]
That would be awesome @gleich 💯 💪

File: extensions/vercast/src/index.tsx [hunk 1]
```diff
@@ -102,3 +123,85 @@ function Main(): JSX.Element {
...
+              <Form.Dropdown.Item 
+                value={JSON.stringify({envId : ev.id, envKey : ev.key})} 
+                title={ev.key} />
+            )
+          })}
+      </Form.Dropdown>
+      <Form.TextArea id="env_value" title="New Value" />
+    </Form>
+  )
+}
```
Comment (gleich) [inline extensions/vercast/src/index.tsx:199]: Would you mind moving this to a new file called `actions.ts`?

File: extensions/vercast/src/index.tsx [hunk 2]
```diff
@@ -102,3 +123,85 @@ function Main(): JSX.Element {
...
+
+function validateForm(values: FormValues): boolean {
+  if (!values.env_value) {
+    showToast(ToastStyle.Failure, "Please set new value");
+    return false;
+  }
+  return true;
+}
```
Comment (gleich) [inline extensions/vercast/src/index.tsx:207]: If this is only being called once then it could just be an anonymous function call.

**Review by gleich — CHANGES_REQUESTED** [review]
Hey @HenriChabrand! Thank you so much for this PR! Left a few suggestions and comments in the review. Nice job getting this done. Really matches the structure of the project. Is there a reason you have some functions that are "raw" that is then only called once by other functions?

File: extensions/vercast/src/index.tsx (hunk 2 above)
Comment (HenriChabrand) [reply extensions/vercast/src/index.tsx:207]: Hello @gleich 👋
Not sure how to do that?

File: extensions/vercast/src/index.tsx (hunk 1 above)
Comment (HenriChabrand) [reply extensions/vercast/src/index.tsx]: Done ✅

**Comment by HenriChabrand** [comment]
Hello @gleich !
I've made some changes to separate the main functionality and the actions.

As for the raw function, I call it only once for now – but I expect some other use cases that might require to use it as well.

**Comment by pernielsentikaer** [comment]
Are we ready to merge this @gleich

File: extensions/vercast/src/index.tsx (hunk 2 above)
Comment (gleich) [reply extensions/vercast/src/index.tsx]: do something like this where the function is being called:

```js
if (!(values) => {
  if (!values.env_value) {
    showToast(ToastStyle.Failure, "Please set new value");
    return false;
  }
  return true;
}) {
  return;
}
```

**Comment by gleich** [comment]
@pernielsentikaer just made one final comment but then we should be good to merge :)

File: extensions/vercast/src/index.tsx (hunk 2 above)
Comment (gleich) [reply extensions/vercast/src/index.tsx]: or you could store the function `validateForm` inside of `UpdateEnvironmentVariable`.

Comment (HenriChabrand) [reply extensions/vercast/src/index.tsx]: I went for the latter and added `validateForm` inside of `UpdateEnvironmentVariable`.

**Comment by HenriChabrand** [comment]
@gleich it should be good now 🙂

**Comment by pernielsentikaer** [comment]
Awesome @HenriChabrand - I'll await feedback from @gleich and then we're gonna merge this ✌️

**Comment by gleich** [comment]
Great! I'll test it and if everything looks good merge it when I get home from school today.

**Comment by PitNikola** [comment]
@gleich, are we good to merge it?

**Comment by gleich** [comment]
yep! Looks good to me :)

**Pull request merged**